lib/core/src/Cardano/Wallet/DB/Model.hs

```diff
@@ -218,13 +224,17 @@ mRollbackTo wid point db = flip (alterModel wid) db $ \wal ->
             then meta { slotId = nearest, status = Pending }
             else meta
 
-    -- | Find nearest checkpoint's slot before or equal to 'point'
-    findNearestPoint :: [Wallet s t] -> SlotId
-    findNearestPoint = head . sortOn Down . mapMaybe fn
+    -- | Find nearest checkpoint's slot before or equal to 'point'.
+    findNearestPoint :: [Wallet s t] -> Maybe SlotId
+    findNearestPoint = headMay . sortOn Down . mapMaybe fn
       where
         fn :: Wallet s t -> Maybe SlotId
         fn cp = if (tip cp <= point) then Just (tip cp) else Nothing
 
+    headMay :: [a] -> Maybe a
```

Comment:
why not name it `headSafe` or `headM` :-)

Reply:
`headSafe` sounds good :) 
The `M` suffix is often used for `Monad` though, so I wouldn't use it here.